include/aws/io/retry_strategy.h

```diff
@@ -0,0 +1,175 @@
+#ifndef AWS_IO_CLIENT_RETRY_STRATEGY_H
+#define AWS_IO_CLIENT_RETRY_STRATEGY_H
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/io/event_loop.h>
+
+#include <aws/common/atomics.h>
+
+struct aws_retry_strategy;
+struct aws_retry_token;
+
+/**
+ * Invoked upon the acquisition, or failure to acquire a retry token. This function will always be invoked if and only
+ * if aws_retry_strategy_acquire_retry_token() returns AWS_OP_SUCCESS. It will never be invoked synchronously from
+ * aws_retry_strategy_acquire_retry_token().
+ */
+typedef void(aws_retry_strategy_on_retry_token_acquired_fn)(
```

Comment:
[clarification] token will always be NULL if error_code is non-zero, and vice-versa, right?
If so mention that in comments

Reply:
will do.